### ISSUE TITLE
Add support for element nuking in web capture.

### ISSUE THREAD

**Issue by burtonator** [burtonator/polar-bookshelf#278]
oh and it would be awesome, to have something like ekill to nuke elements (ads, comments, ..) on websites before save (https://github.com/rhardih/ekill)

**Comment by burtonator** [comment]
What about something like this ... looking at ekill too... 

https://github.com/Simonwep/selection

**Comment by burtonator** [comment]
something like this would work for killing elements in the browers:

```javascript

let targetState = {
    element: undefined,
    restoreOpacity: undefined
}

function restoreTarget() {
    if (targetState.element) {
        console.log("Restopring")


        targetState.element.style.opacity = targetState.restoreOpacity;
    }
}

document.body.addEventListener('mouseover', (event) => {
    console.log(event.target);

    restoreTarget();

    targetState.element = event.target;
    targetState.restoreOpacity = event.target.style.opacity;

    console.log("setting opacity")

    event.target.style.opacity = '0.1';

});

document.body.addEventListener('mouseout', (event) => {

    restoreTarget();

});

document.body.addEventListener('click', (event) => {

    if(targetState.element) {

        targetState.element.parentElement.removeChild(targetState.element);
        targetState.element = undefined;
        targetState.restoreOpacity = undefined;

        event.preventDefault();

    }

});
```

**Comment by setr** [comment]
I think you’re making this more complex than it needs to be: you can already edit the dom (and kill nodes, and much more) with the normal chrome developer tools, when viewing a polar document.

You just can’t save your edits when you’re done. You can save the annotations, but not the html.

Add saving html functionality to the viewer and everything else naturally falls out.

**Comment by burtonator** [comment]
This document:

https://undark.org/2018/10/18/to-sway-skeptics-first-listen/

highlights the problem we have.

The sidebar here needs to be removed.  But it can't be deleted from the DOM to have the content expanded.

Once deleted the side content has a huge padding to the right due to width being set to 65% in the CSS.

Maybe another workaround is to select what SHOULD be capture not what SHOULD NOT ...

I think  I can just have the user select that as part of the DOM and then remove everything else.

**Comment by burtonator** [comment]
... or maybe a combination of the two. The ability to select only a specific node along with the ability to kill nodes.

**Comment by setr** [comment]
I'm thinking it'd be simpler and more powerful to do

1. Leave capture as-is, to simply fetch the webpage
2. Correct the document in the viewer (using developer-tools). Delete node, change text, css etc. Essentially manual post-processing
3. Save over the the document.

I think the ideal would be to have developer-tools working in capture, but I imagine thats not possible with the current system. The second best is to work around it by using the viewer (post-capture), and the **worst-case scenario** is having to re-implement a significantly weaker version of developer-tools in capture.  

Basically I'm of the opinion that you should preserve chromium's dev-tools as the ideal editing mechanism, **by all means possible**. It's by far the best tool for the job.

Alternatively, perhaps have an intermediate stage in capture, by utilizing the viewer? ie change the capture process to:

0. Start a new capture, enter URL
1. capture the webpage, but don't add to polar yet 
      * save to temporary file, or in-memory, as .phz
2. open the page in the viewer 
      * user hits an **edit** button in the capture UI
3. user modifies as he wishes, with dev-tools 
      * already possible with the document-viewer window
      * with the dev-tools window open by default? 
4. save the **edited** webpage to polar (as .phz)
      * this version is what actually gets added to the polar repository
5. Capture window closes

I'm also concerned that if you do a reimplementation of node-deletion, it'll naturally lead to reimplementations of all kinds of editing in the future (like node-deletion often requiring css changes as well, in your example), eventually polluting the polar codebase. 

I believe reimplementation should be avoided as strongly as possible.

**Comment by michivi** [comment]
Hello!

Sorry to barge in like that, but the issue caught my eye as I was thinking about such feature when I discovered the tool (which I find very interesting and useful btw :-) )

Before noticing the issue, I was thinking about using the content extraction method used by [Full-Text RSS](http://fivefilters.org/content-only/): the application's browser would allow the user to directly extract the content of the website and save the simpler HTML version of it (without the aside content, navigation menu etc.). The simple HTML version of a webpage would be created using the rules available in the GitHub repository [ftr-site-config](https://github.com/fivefilters/ftr-site-config). That way, the user will have the choice of which version to save (the full original version, or the simpler one), and s/he may even modify the existing rules. But then, that would bring some other issues: when and how would the rules be updated? what if the user wanted to keep a content that has been removed by the rules? should the original version still be saved just in case?

What do you think? Would it be worth investigating?

**Comment by burtonator** [comment]
I mean for the ftr-site-config it might be interesting to add that as an option alongside readability... so with that and the readability options it might work out well.  

I would still like to have the options I discussed if I can get the UI to work..

**Comment by michivi** [comment]
Yes, I only thought of it as an additional option, perhaps a little easier to implement (because it would apply to many pages easily).

I agree that the element selection and deletion feature through the UI is indeed useful. I might also be interesting to guess the selected elements XPath or associated CSS selection query in order to apply it next time to other pages from the same website.

**Comment by burtonator** [comment]
Instead of kill I think we might be better off with focus:

- click on the DOM.  The element is highlighted by a bounding box. 

- the bounding box has a nav bar at the top. The nav bar allows us to go up and down the DOM.

- once we select the right one and it's focused, we click capture and ONLY that element is saved in the output.
